Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -254,6 +254,25 @@ void CookieStore::get(CookieStoreGetOptions&& options, Ref<DeferredPromise>&& pr
         return;
     }
 
+    if (options.name.isNull() && options.url.isNull()) {
+        promise->reject(ExceptionCode::TypeError);
+        return;
+    }
+
+    if (!options.url.isNull()) {
+        auto parsed = context->completeURL(options.url);
+        if (context->isDocument() && parsed != context->url()) {
+            promise->reject(ExceptionCode::TypeError);
+            return;
+        }
+
+        if (!origin->isSameOriginDomain(SecurityOrigin::create(parsed))) {
+            promise->reject(ExceptionCode::TypeError);
+            return;
+        }
+        options.url = parsed.string();
```

Comment:
According to spec, if `options.url` is null, we should use context's creation url (https://wicg.github.io/cookie-store/#dom-cookiestore-get-options), but I don't see we set options.url to context url.

Reply:
Good catch--fixed!